src/DynamoUtilities/Md2Html.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+
+namespace Dynamo.Utilities
+{
+    internal class Md2Html : IDisposable
+    {
+        private readonly Process process = new Process();
+        private bool started;
+        /// <summary>
+        /// Constructor
+        /// Start the CLI tool and keep it around
+        /// </summary>
+        internal Md2Html()
+        {
+            ProcessStartInfo startInfo = new ProcessStartInfo
+            {
+                CreateNoWindow = true,
+                RedirectStandardOutput = true,
+                RedirectStandardInput = true,
+
+                UseShellExecute = false,
+                Arguments = @"",
+                FileName = GetToolPath()
+            };
+
+            process.StartInfo = startInfo;
+            try
+            {
+                process.Start();
+                started = true;
+            }
+            catch(Win32Exception)
+            {
+                // Do nothing
+            }
+        }
+
+        /// <summary>
+        /// Kill the CLI tool, if still running
+        /// </summary>
+        /// <param name="disposing"></param>
+        protected virtual void Dispose(bool disposing)
+        {
+            KillProcess();
+        }
+
+        /// <summary>
+        /// Kill the CLI tool, if still running
+        /// </summary>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Converts a markdown string into Html.
+        /// </summary>
+        /// <param name="mdString"></param>
+        /// <param name="mdPath"></param>
+        /// <returns>Returns converted markdown as html</returns>
+        internal string ParseMd2Html(string mdString, string mdPath)
+        {
+            if (!started)
+            {
+                return GetErrorMessage();
+            }
+
+            process.StandardInput.WriteLine(@"<<<<<Convert>>>>>");
+            process.StandardInput.WriteLine(mdPath);
+            process.StandardInput.WriteLine(mdString);
+            process.StandardInput.WriteLine(@"<<<<<Eod>>>>>");
+
+            var writer = new StringWriter();
+            GetData(ref writer);
+
+            return writer.ToString();
```

Comment:
So each call to ParseMd2Html is blocking until it receives the response back ?
Is it possible that ParseMd2Html /SanitizeHtmlwill be called from multiple threads ? Can the md2html executable handle concurrent calls ? Are we guarded against such behavior somehow ? (the fact that everything seems to be internal probably helps)

Reply:
The current class is not thread safe so you want to instantiate and call the class from the same thread. But it is OK to have multiple instances. I added some notes about this to the class description. Yes, it is blocking but it is not more blocking than it was in the original implementation. I have seen no issues with that at all the way it currently is used in doc browser UI. So I want to keep this simple approach for now and enhance it later if needed.

Reply:
should we have a timeout for interacting with the md2html.exe ? as a feature later on ..

Reply:
I leave timeout management as a feature for the future but decided to add some exception handling instead. E.g. make sure that we clean up, return an error message and if needed kill the process if we encounter an communication error.